src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,146 @@
+package duke;
+
+import duke.task.Task;
+import java.util.ArrayList;
+
+/**
+ * The class to represent a list of tasks the user has.
+ */
+public class TaskList {
+
+    /** the list of tasks */
+    private ArrayList<Task> listOfTasks;
+
+    /**
+     * The constructor of the TaskList class
+     *
+     * @param tasks the list of tasks
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.listOfTasks = tasks;
+    }
+
+    /**
+     * The constructor of the TaskList class
+     */
+    public TaskList() {
+        this.listOfTasks = new ArrayList<Task>();
+    }
+
+    /**
+     * The method to return the list of tasks
+     *
+     * @return the list of tasks
+     */
+    public ArrayList<Task> getListOfTasks() {
+        return this.listOfTasks;
+    }
+
+    /**
+     * The method to print all tasks to console
+     */
+    public void listAllTasks() {
+        if (listOfTasks.isEmpty()) {
+            System.out.println("You currently have no tasks! Add one now ☻");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            int size = listOfTasks.size();
+            for (int i = 0; i < size; i++) {
+                Task t = listOfTasks.get(i);
+                System.out.println((i + 1) + "." + t.toString());
+            }
+        }
+    }
+
+    /**
+     * Method to determine if the list of tasks is empty
+     *
+     * @return whether the list of tasks is empty
+     */
+    public boolean isEmpty() {
+        return this.listOfTasks.isEmpty();
+    }
+
+    /**
+     * Method to determine the number of elements in the list of tasks
+     *
+     * @return the number of elements in the list of tasks
+     */
+    public int size() {
+        return this.listOfTasks.size();
+    }
+
+    /**
+     * Method to return a task at a specific index in the list
+     *
+     * @return the task at the specific index in the list
+     */
+    public Task get(int i) {
+        return this.listOfTasks.get(i);
+    }
```

Comment:
Should the method name be changed to something that better describes this method?